### PR TITLE
Context change validation

### PR DESCRIPTION
**Pull request by JasBeilin**
## Status
- [x] In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/31190

## Description
Added a validation that checks if the output's context paths in YML matches the README.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by JasBeilin** [comment]
=========== Found errors in the following files ===========

Packs/Pcysys/Integrations/Pcysys/README.md - [RM102]
Packs/Pcysys/Integrations/Pcysys/Pcysys.yml - [IN134]
Packs/Skyformation/Integrations/integration-Skyformation.yml - [IN128]
Packs/ArcSightESM/Integrations/ArcSightESMv2/README.md - [RM102]
Packs/ArcSightESM/Integrations/ArcSightESMv2/ArcSightESMv2.yml - [IN134]
Packs/CrowdStrikeIntel/Integrations/CrowdStrikeFalconIntel_v2/README.md - [RM102]
Packs/CrowdStrikeIntel/Integrations/CrowdStrikeFalconIntel_v2/CrowdStrikeFalconIntel_v2.yml - [IN134]
Packs/RecordedFuture/Integrations/RecordedFuture/README.md - [RM102]
Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.yml - [IN134]
Packs/XMCyber/Integrations/XMCyberIntegration/README.md - [RM102]
Packs/XMCyber/Integrations/XMCyberIntegration/XMCyberIntegration.yml - [IN134]
Packs/XMatters/Integrations/xMatters/README.md - [RM102]
Packs/XMatters/Integrations/xMatters/xMatters.yml - [IN134]
Packs/BastilleNetworks/Integrations/BastilleNetworks/README.md - [RM102]
Packs/BastilleNetworks/Integrations/BastilleNetworks/BastilleNetworks.yml - [IN134]
Packs/TaniumThreatResponse/Integrations/TaniumThreatResponse/TaniumThreatResponse.yml - [IN134]
Packs/Expanse/Integrations/Expanse/README.md - [RM102]
Packs/Humio/Integrations/Humio/README.md - [RM102]
Packs/Humio/Integrations/Humio/Humio.yml - [IN134]
Packs/FireEyeETP/Integrations/FireEyeETP/README.md - [RM102]
Packs/BluelivThreatContext/Integrations/BluelivThreatContext/README.md - [RM102]
Packs/BluelivThreatContext/Integrations/BluelivThreatContext/BluelivThreatContext.yml - [IN134]
Packs/ArcherRSA/Integrations/ArcherV2/README.md - [RM102]
Packs/SecurityAdvisor/Integrations/SecurityAdvisor/README.md - [RM102]
Packs/SecurityAdvisor/Integrations/SecurityAdvisor/SecurityAdvisor.yml - [IN134]
Packs/GooglePubSub/Integrations/GooglePubSub/README.md - [RM102]
Packs/GooglePubSub/Integrations/GooglePubSub/GooglePubSub.yml - [IN134]
Packs/AutoFocus/Integrations/AutofocusV2/README.md - [RM102]
Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.yml - [IN134]
Packs/CortexDataLake/Integrations/CortexDataLake/README.md - [RM102]
Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake.yml - [IN134]
Packs/Infocyte/Integrations/Infocyte/README.md - [RM102]
Packs/Infocyte/Integrations/Infocyte/Infocyte.yml - [IN134]
Packs/Spamcop/Integrations/Spamcop/README.md - [RM102]
Packs/SophosCentral/Integrations/SophosCentral/README.md - [RM102]
Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.yml - [IN134]
Packs/AWS-SecurityHub/Integrations/AWS_SecurityHub/AWS_SecurityHub.yml - [IN134]
Packs/CrowdStrikeMalquery/Integrations/CrowdStrikeMalquery/README.md - [RM102]
Packs/RiskSense/Integrations/RiskSense/README.md - [RM102]
Packs/PrismaCloud/Integrations/RedLock/README.md - [RM102]
Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/README.md - [RM102]
Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.yml - [IN134]
Packs/DNSDB/Integrations/DNSDB_v2/README.md - [RM102]
Packs/Barracuda/Integrations/BarracudaReputationBlockListBRBL/README.md - [RM102]
Packs/Barracuda/Integrations/BarracudaReputationBlockListBRBL/BarracudaReputationBlockListBRBL.yml - [IN134]
Packs/PAN-OS/Integrations/Panorama/README.md - [RM102]
Packs/PAN-OS/Integrations/Panorama/Panorama.yml - [IN134]
Packs/Whois/Integrations/Whois/README.md - [RM102]
Packs/Code42/Integrations/Code42/README.md - [RM102]
Packs/NozomiNetworks/Integrations/NozomiNetworks/NozomiNetworks.yml - [IN134]
Packs/NozomiNetworks/Integrations/NozomiNetworks/README.md - [RM102]
Packs/FeedMitreAttack/Integrations/FeedMitreAttack/README.md - [RM102]
Packs/CortexXDR/Integrations/CortexXDRIR/README.md - [RM102]
Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml - [IN134]
Packs/Polygon/Integrations/Polygon/README.md - [RM102]
Packs/Polygon/Integrations/Polygon/Polygon.yml - [IN134]
Packs/Tufin/Integrations/Tufin/README.md - [RM102]
Packs/Tufin/Integrations/Tufin/Tufin.yml - [IN134]
Packs/AcalvioShadowplex/Integrations/acalvioapp/README.md - [RM102]
Packs/AcalvioShadowplex/Integrations/acalvioapp/acalvioapp.yml - [IN134]
Packs/DeepInstinct/Integrations/DeepInstinct/DeepInstinct.yml - [IN134]
Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/README.md - [RM102]
Packs/CofenseTriage/Integrations/CofenseTriagev2/README.md - [RM102]
Packs/CofenseTriage/Integrations/CofenseTriagev2/CofenseTriagev2.yml - [IN134]
Packs/CounterCraft/Integrations/CounterCraft/README.md - [RM102]
Packs/VMware/Integrations/VMware/README.md - [RM102]
Packs/Gurucul/Integrations/GuruculGRA/README.md - [RM102]
Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml - [IN134]
Packs/Akamai_WAF/Integrations/Akamai_WAF/README.md - [RM102]
Packs/EWS/Integrations/EWSO365/README.md - [RM102]
Packs/Flashpoint/Integrations/Flashpoint/README.md - [RM102]
Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md - [RM102]
Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml - [IN134]
Packs/Uptycs/Integrations/Uptycs/README.md - [RM102]
Packs/Uptycs/Integrations/Uptycs/Uptycs.yml - [IN134]
Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/README.md - [RM102]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Changelog
+* Added a validation that makes sure outputs in integration are matching the README file.
```
Comment (bakatzir) [inline CHANGELOG.md:2]:
```suggestion
* Added a validation that makes sure outputs in integrations are matching the README file.
```

File: demisto_sdk/commands/common/errors.py [hunk 2]
```diff
@@ -67,6 +67,7 @@
     "multiple_default_arg": "IN134",
     "missing_output_context": "IN136",
     "invalid_integration_parameters_display_name": "IN135",
+    "missing_output_context": "IN136",
```
Comment (bakatzir) [inline demisto_sdk/commands/common/errors.py:70]: duplicate, no need

File: demisto_sdk/commands/common/errors.py [hunk 3]
```diff
@@ -443,6 +444,18 @@ def missing_get_mapping_fields_command():
...
+    def readme_missing_output_context(command, context_paths):
+        return f'The Following context paths for command {command} are found in YML file ' \
+               f'but are missing from the README file: {context_paths}'
+
+    @staticmethod
+    @error_code_decorator
+    def missing_output_context(command, context_paths):
+        return f'The Following context paths for command {command} are found in the README file ' \
+               f'but are missing from the YML file: {context_paths}'
+
```
Comment (bakatzir) [inline demisto_sdk/commands/common/errors.py:458]: dup
```suggestion
```

File: demisto_sdk/commands/common/constants.py [hunk 4]
```diff
@@ -1014,3 +1014,5 @@ class PB_Status:
                                      'short_description', 'description', 'Tags', 'blocked']
 
 UUID_REGEX = r'[0-9a-f]{8}\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\b[0-9a-f]{12}'
+
+CONTEXT_OUTPUT_TABLE_HEADER = '| **Path** | **Type** | **Description** |'
```
Comment (bakatzir) [inline demisto_sdk/commands/common/constants.py:1018]:
```suggestion
CONTEXT_OUTPUT_README_TABLE_HEADER = '| **Path** | **Type** | **Description** |'
```

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 5]
```diff
@@ -97,7 +100,9 @@ def is_valid_file(self, validate_rn: bool = True, skip_test_conf: bool = False)
             self.is_valid_max_fetch_and_first_fetch(),
             self.is_valid_as_deprecated(),
             self.is_valid_parameters_display_name(),
-            self.is_mapping_fields_command_exist()
+            self.is_mapping_fields_command_exist(),
+            self.is_context_change_in_readme()
```
Comment (bakatzir) [inline demisto_sdk/commands/common/hook_validations/integration.py:104]:
```suggestion
            self.is_context_change_in_readme(),
```

File: demisto_sdk/commands/common/errors.py (hunk 3 above)
Comment (JasBeilin) [reply demisto_sdk/commands/common/errors.py:460]: I don't see dup in my version- one is for readme and not YML and one is for YML and not readme.

**Review by bakatzir — APPROVED** [review]

**Comment by JasBeilin** [comment]
@roysagi, I am using regex to get the commands and context paths from the readme. I would appreciate it a lot if you can take  a look on it, since it is very non-intuitive.

**Review by bakatzir — APPROVED** [review]

**Pull request merged**